### ISSUE TITLE
Failed to load dynamic library 'libisar.so' - 32-Bit Android 

### ISSUE THREAD

**Issue by spatialbits** [isar/isar#367]
**Steps to Reproduce**

1. Upgrade to 2.5.0.
2. Run application on a physical 32-Bit Android device.
3. Open an Isar instance.
4. Observe error.

```
E/flutter ( 4551): [ERROR:flutter/lib/ui/ui_dart_state.cc(209)] Unhandled Exception: IsarError: Could not initialize IsarCore library. If you create a Flutter app, make sure to add isar_flutter_libs to your dependencies: Invalid argument(s): Failed to load dynamic library 'libisar.so': dlopen failed: cannot locate symbol "__emutls_get_address" referenced by "/data/app/app.example.namevlXoQXzxvZRLxGvHH5jQ7A==/base.apk!/lib/armeabi-v7a/libisar.so"...
```

**Code sample**
- Just open an instance like usual.
```
  final isar = await Isar.open(
    schemas: schemas,
    directory: d.path,
    relaxedDurability: false,
    inspector: kDebugMode,
  );
```

**Version**
 - Platform: Android
 - Device Samsung Galaxy Tab A (2016) T580, 32-Bit Tablet.
 - Flutter version: 2.10.5
 - Isar version: 2.5.0
 - Isar Generator version: 2.5.0

Additional note:
- The error is was observed on:
  - 64-bit virtual pixel device
  - 32-bit physical iphone 7

**Comment by leisim** [comment]
@spatialbits thanks for the report! did it work using 2.4.0?

**Comment by spatialbits** [comment]
It was working in 2.4.0. I've set it back to 2.4.0 and am still getting the same error. 
I had to upgrade the sdk in order to use 2.5.0 so perhaps I bungled something in that process. I'm looking at that now.

**Comment by leisim** [comment]
Thanks for your help with figuring this out 🙌

**Comment by spatialbits** [comment]
The dart and flutter sdks were out of date. I was on flutter 2.8 and upgraded to 2.10.5. Don't recall what dart was on, but it's on 2.16.2 now.

and the below android api and kotlin changes:
![image](https://user-images.githubusercontent.com/4192023/166335489-58b5f355-0a51-4150-aa58-6d53e576e0f8.png)

**Comment by spatialbits** [comment]
ok actually I had confused myself when switching between 2.4.0 and 2.5.0, but I have that sorted. Keeping all dependencies the same and setting only isar-related packages I can  the problem.
![image](https://user-images.githubusercontent.com/4192023/166337639-01ef2dca-61e2-427e-933c-be196ed6263e.png)

flutter doctor in case it helps:
```
    • Flutter version 2.10.5 at /Users/user/dev/flutter
    • Upstream repository ssh://git@github.com/flutter/flutter.git
    • Framework revision 5464c5bac7 (2 weeks ago), 2022-04-18 09:55:37 -0700
    • Engine revision 57d3bac3dd
    • Dart version 2.16.2
    • DevTools version 2.9.2

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0-rc5)
    • Android SDK at /Users/user/Android/sdk
    • Platform android-31, build-tools 31.0.0-rc5
    • ANDROID_HOME = /Users/user/Library/Android/sdk
    • ANDROID_SDK_ROOT = /Users/user/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.11+0-b60-7772763)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 13.3.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • CocoaPods version 1.11.2

[✓] Android Studio (version 2021.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.11+0-b60-7772763)

[✓] VS Code (version 1.66.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.40.0

[✓] Connected device (3 available)
    • SM T580 (mobile)                     • 52031124538864f1                         • android-arm   • Android 8.1.0 (API 27)
    • Android SDK built for arm64 (mobile) • emulator-5554                            • android-arm64 • Android 9 (API 28) (emulator)
    • iPhone (mobile)                      • 9f1fb333dd5ffd436d999c73e81b4fec25cd456e • ios           • iOS 15.4.1 19E258

[✓] HTTP Host Availability
    • All required HTTP hosts are available
```

**Comment by richard457** [comment]
It's not working for me too!

**Comment by spatialbits** [comment]
@richard457 make sure you revert your lock file and flutter clean.

**Comment by jpolstre** [comment]
It still doesn't work on 32 bit Android, for me either.

**Comment by leisim** [comment]
This should be fixed in `3.0.0-dev.0` 🙂

**Comment by jpolstre** [comment]
it works 🎉!!!, thanks.
